source/lex.h

```diff
@@ -825,29 +825,61 @@ auto lex_line(
                     store(3, lexeme::Not);
                 }
 
+                //G
                 //G decimal-literal:
-                //G     digit { ' | digit }*
+                //G     digit { ' | digit }* [uU][lL][lL]
                 //G
                 //G floating-point-literal:
-                //G     digit { ' | digit }* . digit { ' | digit }*
-                //GTODO full grammar
+                //G     digit { ' | digit }* (. digit { ' | digit }*)? ([eE][-+]?digit { ' | digit }*) [fFlL]
```

Comment:
I use a regex-like syntax for the grammar definition here, a bit sloppy, if there's a reference on the notation you prefer to use I'm happy to update it.